### ISSUE TITLE
Cannot install user certificates in the Settings app

### ISSUE THREAD

**Issue by george-n-young** [GrapheneOS/os-issue-tracker#1340]
It is currently not possible to add your own certificate authorities to GrapheneOS.

Steps to reproduce:

1. Go to Settings -> Security -> Encryption and Credentials -> "Install a certificate"
3. Press "CA Certificate"
4. Press "Install anyway"
5. Choose the desired .crt file

After completing step 5, it quickly returns the user back to the "Install a certificate" page. **No errors are displayed, and yet the certificate remains uninstalled.** 

The certificate is no where to be found when looking through the "User credentials" page, the "Trusted credentials" page, or anywhere else. The ability to add VPN certificates and WiFi certificates is also broken and leads to the same error.

I am using GrapheneOS Android 12 on an Pixel 6 Pro (no root).

**Comment by thestinger** [comment]
It's likely invalid in some way. Support for user added CA certificates has been heavily restricted and largely removed in modern Android. It isn't anything to do with GrapheneOS specifically. I don't think it's possible to add a CA for Wi-Fi anymore. They're supposed to get a proper certificate for it.

**Comment by thestinger** [comment]
Also, this isn't how you do VPN certificates.

**Comment by george-n-young** [comment]
> I don't think it's possible to add a CA for Wi-Fi anymore.

Ahh okay then, if this is the case perhaps it makes sense to remove the page to add a CA authority all together?

I don't believe the problem is with my .crt file as it is working on two of my other devices.

**Comment by thestinger** [comment]
@george-n-young Are they Android 12? There are much more restrictions on Android 11 and 12.

**Comment by george-n-young** [comment]
Makes sense, alright. Any chance either of you know of a good guide online that can help me figure out how to make my .crt file compatible with Android 12? Very new to this sort of thing.
